account_analytic_parent/models/account_analytic_account.py

```diff
@@ -59,3 +59,9 @@ def name_get(self):
                 current = current.parent_id
             res.append((account.id, name))
         return res
+
+    @api.multi
+    def write(self, vals):
+        if 'active' in vals:
+            self.mapped('child_ids').toggle_active()
+        return super(AccountAnalyticAccount, self).write(vals)
```

Comment:
I think it should be this way:
```python
    @api.multi
    def write(self, vals):
        if 'active' in vals and not vals['active']:
            self.mapped('child_ids').active = False
        return super(AccountAnalyticAccount, self).write(vals)
```
because activating the parent doesn't mean you want active all its children

Reply:
edited

Reply:
It has to be with toggle_active because calling write will get recursion error.

Reply:
I don't see the recursion error

Reply:
I am not just saying, I tested that.

Reply:
ok, then do instead:

```python

    @api.multi
    def archive_accounts(self):
        for account in self:
            record.active = False

    @api.multi
    def write(self, vals):
        if 'active' in vals and not vals['active']:
            self.mapped('child_ids').archive_accounts()
        return super(AccountAnalyticAccount, self).write(vals)
```

Reply:
yes, I forgot to change also that.  Thanks.